user/mkdir.c

```diff
@@ -1,23 +1,78 @@
 #include "kernel/types.h"
 #include "kernel/stat.h"
+#include "Kernel/fcntl.h"
 #include "user/user.h"
 
-int
-main(int argc, char *argv[])
+int makeDirectory (char *path){
+  char *p;
+  char temp;
+  int status = 0;
+
+  // Skip leading slashes
+  p = path;
+  while (*p == '/')
+    p++;
+
+  while (*p != 0) {
+    // Find the next component
+    while (*p != '/' && *p != 0)
+      p++;
+
+    temp = *p;
+    *p = 0; // Temporarily clear string value
+
+    if (mkdir(path) < 0) {
+      // Atempt to open the directory
+      int fd = open(path, O_RDONLY); // fd = file descriptor
```

Comment:
minor: not sure though. on a quick glance on `man mkdir` `-p` should be fine if the parent directory already exist i think we can remove the conditional.

Reply:
suppose we do wanna consider error handling, we should consider fd >= 0 as (highly unlikely) 0 stdin is valid fd as well.